client/server/pages/index.js

```diff
@@ -693,19 +695,21 @@ function wpcomPages( app ) {
 		}
 	} );
 
-	app.get( '/plans', function ( req, res, next ) {
+	const lang = getLanguageRouteParam( 'locale' );
+	app.get( `/${ lang }/plans`, function ( req, res, next ) {
 		if ( ! req.context.isLoggedIn ) {
 			const queryFor = req.query?.for;
 			const ref = req.query?.ref;
+			const locale = req.params?.locale;
 
 			if ( queryFor && 'jetpack' === queryFor ) {
 				res.redirect(
 					'https://wordpress.com/wp-login.php?redirect_to=https%3A%2F%2Fwordpress.com%2Fplans'
 				);
 			} else {
-				const pricingPageUrl = ref
-					? `https://wordpress.com/pricing/?ref=${ ref }`
-					: 'https://wordpress.com/pricing/';
+				const pricingPage = 'https://wordpress.com/pricing/';
+				const refQuery = ref ? `?ref=${ ref }` : '';
+				const pricingPageUrl = localizeUrl( `${ pricingPage }${ refQuery }`, locale );
 				res.redirect( pricingPageUrl );
 			}
 		} else {
```

Comment:
Not sure if the redirect from `/:locale/plans` to `/plans` for [logged in users](https://github.com/Automattic/wp-calypso/pull/79562/commits/3ce6a682ddfa1babf6e62acc7343926194ee152c#diff-ef20b8cf3c49260b6a20f1391f446478fb09f05d43089d2ce2dba4d09c9a86d8L715-L717) has been removed intentionally, but I think we still need it.